api/commands/build.ts

```diff
@@ -20,28 +19,31 @@ export async function build(
   runConfig: dataform.IRunConfig,
   dbadapter: dbadapters.IDbAdapter
 ) {
-  const prunedGraph = prune(compiledGraph, runConfig);
-  const stateResult = await state(prunedGraph, dbadapter);
```

Comment:
As discussed - I think this could have a significant impact on performance of runs during development, e.g. projects with 500+ tables which will need to fetch a TON of metadata before doing anything.

Makes sense to remove the pruning code which was confusing, but perhaps the state computation should only fetch table metadata for the set of all transitive inputs + included actions to keep it snappy.